tcms/rpc/api/testexecution.py

```diff
@@ -183,12 +183,21 @@ def update(execution_id, values, **kwargs):
 
     # if this call updated TE.status then adjust timestamps
     if values.get("status"):
+        now = timezone.now()
         if test_execution.status.weight != 0:
-            test_execution.close_date = timezone.now()
+            test_execution.close_date = now
         else:
             test_execution.close_date = None
         test_execution.save()
 
+        all_executions = TestExecution.objects.filter(run=test_execution.run)
+        if test_execution.status.weight != 0 and not all_executions.filter(status__weight=0).exists():
+            test_execution.run.stop_date = now
+            test_execution.run.save()
+        elif test_execution.status.weight == 0 and not test_execution.run.stop_date:
```

Comment:
> ok, but when I tested this locally it did not achieved what we wanted it to (or I don't understand what we want to achieve)

What we want is to change `stop_date` only if necessary which is when TR.stop_date is already set and one of the TE changes its status to weight 0. 

The `not` part in the `and not` condition above is the issue. I've typed it incorrectly the first time around and you have copied it verbatim.

Reply:
I did not copied your suggestion, I tried to make sense of it, but couldn't understand why we need any check at all. looks like the answer is Save hooks and now it makes sense, but before that it didn't with or without the `not`.